### ISSUE TITLE
Recording line is lagging behind

### ISSUE THREAD

**Issue by HarelM** [IsraelHikingMap/Site#1749]
## What I expect to happen
Recoding line should always be where the GPS is (when the velocity and accuracy are correct)

## What really happened
Recoding line is only updated when the map is done moving to the new GPS position

## What I did that caused the issue - step by step
1. Start recoding
2. Zoomed in a lot
3. Looked at the orange recording line

## My environment
* Operating System and version:
* Web Browser and version:

## Things I think the developers should know, images, links etc.
This is related somehow to redux, zone, and capacitor/cordova plugin relations.
I'm failing to understand what's causing this.
The bottom line, after I added some tracing is that the routes in routes.component are updated 3 times for each new GPS point (which is weird):
1. When the recording arrives - point is added
2. When the GPS position is updated - routes is using the previous state and "removes" the added point in the line
3. When the map is updated - when the map stops moving and routes are "OK" again.

**Comment by HarelM** [comment]
More info into this issue in case I read it in the future:
The reason for this is that when you change the store and listening to a store change and then dispatching a new action to change the store is the core problem here.
Getting a point from the GPS -> update state -> listen to it in the recoding service -> add point to recorded route -> update state <-- problem...
